src/server.rs

```diff
@@ -166,6 +191,20 @@ impl ServerStats {
             }};
         }
 
+        macro_rules! set_duration_stat {
+            ($vec:ident, $dur:expr, $num:expr, $name:expr) => {{
+                let mut stat = CacheStatistic::new();
+                stat.set_name(String::from($name));
+                if $num > 0 {
+                    let duration = $dur / $num as u32;
+                    stat.set_str(format!("{}.{:03} s", duration.as_secs(), duration.subsec_nanos() / 1000_000));
```

Comment:
I really wish `Duration` had an `as_fractional_secs` method or something like that. [I filed an issue on that](https://github.com/rust-lang/rust/issues/38475). Also: oops on the conversion bug. I wish there was at least a constant in the stdlib we could use for it!